tests/tests_e2e/az_ulib_ipc_e2e/az_ulib_ipc_e2e.c

```diff
@@ -0,0 +1,542 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+#include <stddef.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "az_pal_os_api.h"
+#include "az_ulib_action_api.h"
+#include "az_ulib_descriptor_api.h"
+#include "az_ulib_ipc_api.h"
+#include "azure_macro_utils/macro_utils.h"
+#include "test_thread.h"
+#include "testrunnerswitcher.h"
+#include "ulib_result.h"
+#include "umock_c/umock_c.h"
+
+static TEST_MUTEX_HANDLE g_test_by_test;
+
+MU_DEFINE_ENUM_STRINGS(UMOCK_C_ERROR_CODE, UMOCK_C_ERROR_CODE_VALUES)
+
+static void on_umock_c_error(UMOCK_C_ERROR_CODE error_code) {
+  ASSERT_FAIL("umock_c reported error :%s", MU_ENUM_TO_STRING(UMOCK_C_ERROR_CODE, error_code));
+}
+
+static uint32_t my_property = 0;
+
+static AZ_ULIB_RESULT get_my_property(const void *model_out) {
+  uint32_t *new_val = (uint32_t *)model_out;
+
+  *new_val = my_property;
+
+  return AZ_ULIB_SUCCESS;
+}
+
+static AZ_ULIB_RESULT set_my_property(const void *const model_in) {
+  uint32_t *new_val = (uint32_t *)model_in;
+
+  my_property = *new_val;
+
+  return AZ_ULIB_SUCCESS;
+}
+
+typedef struct my_method_model_in_tag {
+  uint8_t action;
+  const az_ulib_interface_descriptor *descriptor;
+  az_ulib_ipc_interface_handle handle;
+  az_ulib_action_handle action_handle;
+  AZ_ULIB_RESULT return_result;
+} my_method_model_in;
+
+static AZ_ULIB_RESULT my_method(const void *const model_in, const void *model_out) {
+  my_method_model_in *in = (my_method_model_in *)model_in;
+  AZ_ULIB_RESULT *result = (AZ_ULIB_RESULT *)model_out;
+  my_method_model_in in_2;
+
+  switch (in->action) {
+    case 0:
```

Comment:
>0 [](start = 9, length = 1)

consider making these action possibilities #defines to be more explicit of what they're doing